addons/controllers/clusterbootstrap_controller.go

```diff
@@ -203,7 +245,7 @@ func (r *ClusterBootstrapReconciler) createOrPatchclusterBootstrapFromTemplate(c
 	}
 
 	clusterBootstrapTemplate := &runtanzuv1alpha3.ClusterBootstrapTemplate{}
-	key := client.ObjectKey{Namespace: constants.TKGSystemNS, Name: tkrName}
+	key := client.ObjectKey{Namespace: r.Config.AddonNamespace, Name: tkrName}
```

Comment:
Nit: Not sure if `AddonNamespace` is a good name here.

Reply:
no, it is not a good name. Switching to a different config struct will address this.